demo/kserve/custom-manifests/caikit/caikit-standalone/caikit-standalone-isvc-grpc-template.yaml

```diff
@@ -0,0 +1,56 @@
+apiVersion: networking.istio.io/v1alpha3 # workaround for https://issues.redhat.com/browse/RHOAIENG-165
+kind: EnvoyFilter
+metadata:
+  name: grpc-only
+spec:
+  workloadSelector:
+    labels:
+      procotol: "grpconly"
+  priority: 20
+  configPatches:
+    - applyTo: HTTP_FILTER
+      match:
+        context: SIDECAR_INBOUND
+        listener:
+          filterChain:
+            filter:
+              name: envoy.filters.network.http_connection_manager
+      patch:
+        operation: INSERT_BEFORE
+        value:
+          name: envoy.filters.http.lua
+          typed_config:
+            "@type": type.googleapis.com/envoy.extensions.filters.http.lua.v3.Lua
+            inlineCode: |
+              function envoy_on_request(request_handle)
+                local header_value = request_handle:headers():get("Content-Type")
+                local pattern = "^application/grpc"
```

Comment:
question: if I'm sending a HTTP request to a gRPC service, I may not be setting the Content-type to application/grpc because I might be thinking of having an HTTP service

Reply:
The idea is that you know that you are targeting a model via grpc or model serving metadata should have that somehow. When user uses a model endpoint should know what it serves in general. This envoy filter should block anything that does not match the header.
The reason the header should be supplied is [this](https://chromium.googlesource.com/external/github.com/grpc/grpc/+/HEAD/doc/PROTOCOL-HTTP2.md):
> If Content-Type does not begin with “application/grpc”, gRPC servers SHOULD respond with HTTP status of 415 (Unsupported Media Type). This will prevent other HTTP/2 clients from interpreting a gRPC error response, which uses status 200 (OK), as successful.

At least that is the idea of how to detect grpc in the first place.

Reply:
got the point, but based on my attempts, a cURL request with application/grpc header doesn't break the model server, but it gets broken with a cURL request with application/json header (or omitting the -H)

In our specific case, one problem is that RHOAI/ODH Dashboard report the endpoint as it was an HTTP one, hence it could induce users using a json cURL request without setting the grpc header.

Reply:
>It gets broken with a cURL request using application/json header (or omitting the -H)

This should not happen afaik. Probably envoy filter is not applied.